src/components/OSMElementData/OSMElementData.jsx

```diff
@@ -0,0 +1,196 @@
+import { Fragment, useState, useEffect } from 'react'
+import PropTypes from 'prop-types'
+import { FormattedMessage, injectIntl }
+       from 'react-intl'
+import _map from 'lodash/map'
+import _compact from 'lodash/compact'
+import _isEmpty from 'lodash/isEmpty'
+import _sortBy from 'lodash/sortBy'
+import _find from 'lodash/find'
+import _get from 'lodash/get'
+import _flatten from 'lodash/flatten'
+import AsMappableTask
+       from '../../interactions/Task/AsMappableTask'
+import AsIdentifiableFeature
+       from '../../interactions/TaskFeature/AsIdentifiableFeature'
+import Dropdown from '../Dropdown/Dropdown'
+import SvgSymbol from '../SvgSymbol/SvgSymbol'
+import BusySpinner from '../BusySpinner/BusySpinner'
+import messages from './Messages'
+
+const OSM_SERVER = window.env.REACT_APP_OSM_SERVER
+
+const OSMElementData = props => {
+  const [selectedFeatureId, setSelectedFeatureId] = useState(null)
+  const [element, setElement] = useState(null)
+  const [fetchingElement, setFetchingElement] = useState(null)
+  const [failedElement, setFailedElement] = useState(null)
+  const [fetchErr, setFetchErr] = useState(null)
+
+  const { fetchOSMElement, taskBundle } = props
+  const primaryTask = props.task
+  const allTasks = taskBundle ? taskBundle.tasks : [primaryTask]
+
+  const featureIds = _flatten(_compact(_map(allTasks, task => {
+    const geometries = AsMappableTask(task).normalizedGeometries()
+    return geometries ? _compact(_map(
+      geometries.features,
+      f => AsIdentifiableFeature(f).normalizedTypeAndId(true, '/')
+    )) : null
+  })))
+
+  // If the selected feature is no longer available (e.g. because a task bundle
+  // was unbundled), then clear the selection
+  useEffect(() => {
+    if (selectedFeatureId && featureIds.indexOf(selectedFeatureId) === -1) {
+      setSelectedFeatureId(null)
+    }
+  }, [featureIds, selectedFeatureId])
+
+  // Fetch and process the OSM Data
+  useEffect(() => {
+    const activeFeatureId = selectedFeatureId ? selectedFeatureId : featureIds[0]
+
+    // If we're already fetching data for the active feature, or if the fetch failed,
+    // there's nothing to do
+    if (fetchingElement === activeFeatureId || failedElement === activeFeatureId) {
+      return
+    }
+
+    else if (_isEmpty(featureIds)) {
+      // TODO: should I call setElement(null)?
+      // No features to fetch, so nothing to do
+      return
+    } else if (element) {
+      return
+    }
```

Comment:
This is wrong as it will not reload if the id changes. But this is called in a loop without it, so I may need memoization.